site/docs/cow-and-mor.md

```diff
@@ -0,0 +1,195 @@
+2. **partition equality deletes**: convert input predicate \[3\] to partition equality predicates and write partition equality deletes
+3. **partition global deletes**: convert input predicate to equality predicates and write global equality deletes 
+
+\[1\] scan here can mean a table scan, or a scan of unflushed files (stored in memory, local RocksDB, etc.) for use cases like streaming
+
+\[2\] it is in theory possible to write global position deletes, but the writer already knows the exact partitions to write, so it is almost always preferred to write partition position deletes because it costs the same and improves the efficiency of the MoR read process.
+
+\[3\] if the input inequality predicate cannot be converted to a finite number of equality predicates (e.g. `price > 2.33`), then it is only possible to use position deletes instead.
+
+### Data File Reader with Deletes
+
+During MoR read time, the Iceberg reader indexes all the delete files and determines the associated delete files for each data file. Typically,
+
+- as described before, delete files are only applied to data files that has a strictly lower sequence number
+- global deletes are associated with every data file, so it is desirable to have as few global deletes as possible
+- partition deletes are pruned based on their statistics and secondary index information so that each data file is associated with the minimum number of necessary delete files possible.
+
+Because position deletes must be sorted by file path and row position, applying position deletes to data files can be done by streaming the rows in position deletes.
+Therefore, there is not too much burden on memory side, but the number of IO increases as the number of position delete files increases, so it's desirable to have a low number of position deletes for each data file.
+
+For equality deletes, all the equality predicates must be loaded to memory to check against every row in data files.
+Therefore, having too many equality deletes can have the risk of running out of memory.
+It is possible to use an external storage like RocksDB to store equality predicates, but too many equality deletes would still have huge impact to the read performance.
+
+Because of this difference, position deletes tend to be more efficient in reading than equality deletes.
+This has an impact to our delete compaction strategies.
+
+### MoR delete compaction
+
+After being able to read and write delete files, users need to run delete compactions to optimize the read performance.
+There are 3 common Iceberg delete compactions:
+
+#### Rewrite position deletes (REWRITE)
+
+As we discussed in the [reader](#data-with-delete-reader) section, having fewer position deletes per data file is computationally more efficient.
+This compaction optimizes the positions deletes layout, such as combining all position deletes in a single partition to a single file.
```

Comment:
Something about this sentence I can't put my finger on, but I would maybe restate it as:
"This compaction includes optimizations on the position deletes layout, such as combining all position deletes for an individual partition into a single file."